lib/ansible/modules/cloud/vmware/vmware_host_tcpip_stacks.py

```diff
@@ -0,0 +1,473 @@
+                # netstack_spec.netStackInstance.key.ipRouteConfig.ipV6GatewayDevice
+            else:
+                netstack_spec.netStackInstance.ipRouteConfig.defaultGateway = vmkernel_gateway
+                # netstack_spec.netStackInstance.key.ipRouteConfig.gatewayDevice
+        if instance.requestedMaxNumberOfConnections != num_connections:
+            changed = True
+            num_connections_previous = instance.requestedMaxNumberOfConnections
+            netstack_spec.netStackInstance.requestedMaxNumberOfConnections = num_connections
+        if instance.congestionControlAlgorithm != congestion_algorithm:
+            changed = True
+            congestion_algorithm_previous = instance.congestionControlAlgorithm
+            netstack_spec.netStackInstance.congestionControlAlgorithm = congestion_algorithm
+        # Routes can only be defined for custom stacks
+        # netstack_spec.netStackInstance.routeTableConfig = vim.host.IpRouteTableConfig()
+        return netstack_spec, changed, \
+            vmkernel_gateway_previous, num_connections_previous, congestion_algorithm_previous
+
+
+def main():
+    """Main"""
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(
+        esxi_hostname=dict(required=False, type='str'),
+        cluster_name=dict(required=False, type='str'),
+        default=dict(
+            type='dict',
+            options=dict(
+                vmkernel_gateway=dict(type='str'),
+                congestion_algorithm=dict(type='str', default='newreno', choices=['newreno', 'cubic']),
+                num_connections=dict(type='int', default=11000),
+            ),
+        ),
+        provisioning=dict(
+            type='dict',
+            options=dict(
+                vmkernel_gateway=dict(type='str'),
+                congestion_algorithm=dict(type='str', default='newreno', choices=['newreno', 'cubic']),
+                num_connections=dict(type='int', default=11000),
+            ),
+        ),
+        vmotion=dict(
+            type='dict',
+            options=dict(
+                vmkernel_gateway=dict(type='str'),
+                congestion_algorithm=dict(type='str', default='newreno', choices=['newreno', 'cubic']),
+                num_connections=dict(type='int', default=11000),
+            ),
+        ),
+        vxlan=dict(
+            type='dict',
+            options=dict(
+                vmkernel_gateway=dict(type='str'),
+                congestion_algorithm=dict(type='str', default='newreno', choices=['newreno', 'cubic']),
+                num_connections=dict(type='int', default=11000),
+            ),
+        ),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        required_one_of=[
+            ['cluster_name', 'esxi_hostname'],
+        ],
+        mutually_exclusive=[
+            ['cluster_name', 'host_name'],
+        ],
+        supports_check_mode=True
+    )
+
+    dns = VmwareHostTcpIpStacks(module)
```

Comment:
Please change the `dns` to something more suitable for the module as variable name

Reply:
```suggestion
    VmwareHostTcpIpStacks(module).ensure()
```

Reply:
It might be, but all other modules do it like this. I used the `vmware_host_dns` module to create this module...

Reply:
@ckotte , i'm definitely fine with the variable, the only thing i suggest is to have a variable name that is in line with the module (i understand where the dns comes from, it's a small detail, but it helps in clean code)

Reply:
Sure. I will fix it. Thanks for the review